### PR TITLE
refactor: missing function, query method

### PR DESCRIPTION
**Pull request by yogip07**
## 📑 Deskripsi
Menyesuaikan missing function dan query pada modul auth

## ✅ Checks
- [v ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [v] All the tests have passed

## ℹ Informasi Tambahan

### CODE REVIEW

**Comment by WahidinAji** [comment]
walahhh, tiwas aku ubah wkwkwk. yowes gpp. lanjutin aja. btw, querynya dipindah ke goose ya.

terus, jangan lupa gabit tinyint ke smallint. karena pgsql ga ada tinyint. gatau ini bapack @aldy505 kok bisa tinyint ;D

**Comment by yogip07** [comment]
> walahhh, tiwas aku ubah wkwkwk. yowes gpp. lanjutin aja. btw, querynya dipindah ke goose ya.
> 
> terus, jangan lupa gabit tinyint ke smallint. karena pgsql ga ada tinyint. gatau ini bapack @aldy505 kok bisa tinyint ;D

apanya yg kamu ubah Ji? kayaknya aku ga ngubah-ngubah file migration

File: auth/service/user.go [hunk 1]
```diff
@@ -13,26 +13,17 @@ import (
...
-	}()
-
-	tx, err := conn.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSnapshot})
+	tx, err := d.pool.Begin(ctx)
 	if err != nil {
 		return 0, fmt.Errorf("starting transaction: %w", err)
 	}
 
-	result, err := tx.ExecContext(
+	var insertId int64
```
Comment (WahidinAji) [inline auth/service/user.go]: ini hapus aja,

File: auth/service/user.go [hunk 2]
```diff
@@ -48,7 +39,9 @@ func (d *AuthService) CreateUser(ctx context.Context, user *auth.User) (id int64
 				updated_by
 			)
 			VALUES
-			($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)`,
+			($1, $2, $3, $4, $5, $6, $7, $8, $9, $10)
+			RETURNING
```
Comment (WahidinAji) [inline auth/service/user.go]:
```suggestion
			RETURNING id
```

File: auth/service/user.go [hunk 3]
```diff
@@ -59,27 +52,18 @@ func (d *AuthService) CreateUser(ctx context.Context, user *auth.User) (id int64
 		time.Now(),
 		time.Now(),
 		"system",
-	)
+	).Scan(&insertId)
```
Comment (WahidinAji) [inline auth/service/user.go]:
```suggestion
	).Scan(&id)
```

ini pake named return, bisa manfaatin itu

File: auth/service/user.go [hunk 4]
```diff
@@ -13,26 +13,17 @@ import (
...
-	}
-	defer func() {
-		err := conn.Close()
-		if err != nil {
-			log.Printf("error closing connection: %v", err)
-		}
-	}()
-
-	tx, err := conn.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSnapshot})
+	tx, err := d.pool.Begin(ctx)
```
Comment (WahidinAji) [inline auth/service/user.go]: beginTx, kasih IsoLevel. pake ReadCommitted, tambahin `AccessMode: pgx.ReadOnly`
```suggestion
	tx, err := d.pool.BeginTx(ctx, pgx.TxOptions{IsoLevel: pgx.ReadCommitted, AccessMode: pgx.ReadWrite})
```

**Review by WahidinAji — CHANGES_REQUESTED** [review]

File: auth/service/repository.go [hunk 5]
```diff
@@ -27,40 +27,32 @@ func (d *AuthService) CreateUserRepository(ctx context.Context, userId int64, re
...
-		}
-
-		return fmt.Errorf("failed to prepare query: %w", err)
-	}
+	VALUES
+	($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14)`
 
 	for _, repository := range repositories {
-		_, err := tx.StmtContext(ctx, query).ExecContext(
+		_, err := tx.Exec(
```
Comment (WahidinAji) [inline auth/service/repository.go:62]: di pgx juga ada prepare btw pack,
see https://github.com/jackc/pgx/blob/4ebf1d2e0baa87ab6afbbf6a0947c9fb8a30e9d4/conn_test.go#L384

File: auth/service/user.go (hunk 4 above)
Comment (yogip07) [reply auth/service/user.go]: `ReadWrite` maksudnya?

Comment (WahidinAji) [reply auth/service/user.go]: iya

File: auth/service/repository.go (hunk 5 above)
Comment (yogip07) [reply auth/service/repository.go:62]: oh ok. btw kenapa harus pakai prepare?

Comment (WahidinAji) [reply auth/service/repository.go:62]: if there're 5 data
with prepare
```sql
prepare anu insert bla,bla,bla

for i range data
  exec.anu(i.bla,i.bla2,i.bla3)

without prepare
for i range data
   inser bla1,bla2,bla3
```

lebih sederhaanya

```sql
bapak lebih pilih 
insert into table(bla,bla,bla)
insert into table(bla,bla,bla)
insert into table(bla,bla,bla)
insert into table(bla,bla,bla)
insert into table(bla,bla,bla)

atau
inser into table(
(bla,bla2,bla3),
(bla,bla2,bla3),
(bla,bla2,bla3),
(bla,bla2,bla3),
(bla,bla2,bla3),
);
```
terus, kalau penasaran lagi kenapa.

coba bapak benchmark di yugabytenya langsung,
if insert with/without prepare.

**Comment by yogip07** [comment]
done at [410399843dc1](https://github.com/kodiiing/core/pull/19/commits/e60bbed273937567e4f33b9a97e434b019081ea1) check lagi ya ji

**Comment by WahidinAji** [comment]
> walahhh, tiwas aku ubah wkwkwk. yowes gpp. lanjutin aja. btw, querynya dipindah ke goose ya.
> 
> terus, jangan lupa gabit tinyint ke smallint. karena pgsql ga ada tinyint. gatau ini bapack @aldy505 kok bisa tinyint ;D

migrationnya pack

**Review by aldy505 — APPROVED** [review]

**Comment by yogip07** [comment]
i changed `timestamp` into `timestamptz` at https://github.com/kodiiing/core/pull/19/commits/7f6eb59ed4d05fc8412fc3fa173dc874a5201e5d and https://github.com/kodiiing/core/pull/19/commits/22667ec703b190a48922f0470082e70024f1d72d

@aldy505

**Review by WahidinAji — APPROVED** [review]
GBTM

**Pull request merged**